truffleHogger/truffleHogger.py

```diff
@@ -302,60 +302,98 @@ def clone_git_repo(git_url):
 
-    print("~~~~~~~~~~~~~~~~~~~~~")
-    reason = "{}Reason: {}{}".format(bcolors.OKGREEN, reason, bcolors.ENDC)
-    print(reason)
-    dateStr = "{}Date: {}{}".format(bcolors.OKGREEN, commit_time, bcolors.ENDC)
-    print(dateStr)
-    hashStr = "{}Hash: {}{}".format(bcolors.OKGREEN, commitHash, bcolors.ENDC)
-    print(hashStr)
-    filePath = "{}Filepath: {}{}".format(bcolors.OKGREEN, path, bcolors.ENDC)
-    print(filePath)
+    reason = f"{bcolors.OKGREEN}Reason: {reason}{bcolors.ENDC}"
+    date_str = f"{bcolors.OKGREEN}Date: {commit_time}{bcolors.ENDC}"
+    hash_str = f"{bcolors.OKGREEN}Hash: {commit_hash}{bcolors.ENDC}"
+    file_path = f"{bcolors.OKGREEN}Filepath: {path}{bcolors.ENDC}"
+    lines_str = f"{bcolors.OKGREEN}Lines: {lines_found}{bcolors.ENDC}"
+
+    if mask_secrets:
+        detail_str = f"{bcolors.OKGREEN}DetailedLines: <masked-possible-passwords> {bcolors.ENDC}"
+    else:
+        detail_str = f"{bcolors.OKGREEN}DetailedLines: {detailed_found}{bcolors.ENDC}"
 
     if sys.version_info >= (3, 0):
-        branchStr = "{}Branch: {}{}".format(bcolors.OKGREEN, branch_name, bcolors.ENDC)
-        print(branchStr)
-        commitStr = "{}Commit: {}{}".format(bcolors.OKGREEN, prev_commit, bcolors.ENDC)
-        print(commitStr)
-        if print_diff:
-            print(printableDiff)
+        branch_str = f"{bcolors.OKGREEN}Branch: {branch_name}{bcolors.ENDC}"
+        commit_str = f"{bcolors.OKGREEN}Commit: {prev_commit}{bcolors.ENDC}".replace('\n', '')
+        diff = printable_diff if print_diff else '<suppressed>'
+        diff_str = f'{bcolors.OKGREEN}Diff: {diff}{bcolors.ENDC}'
     else:
-        branchStr = "{}Branch: {}{}".format(bcolors.OKGREEN, branch_name.encode('utf-8'), bcolors.ENDC)
-        print(branchStr)
-        commitStr = "{}Commit: {}{}".format(bcolors.OKGREEN, prev_commit.encode('utf-8'), bcolors.ENDC)
-        print(commitStr)
-        if print_diff:
-            print(printableDiff.encode('utf-8'))
-    print("~~~~~~~~~~~~~~~~~~~~~")
+        branch_str = f"{bcolors.OKGREEN}Branch: {branch_name.encode('utf-8')}{bcolors.ENDC}"
+        commit_str = f"{bcolors.OKGREEN}Commit: {prev_commit.encode('utf-8')}{bcolors.ENDC}".replace('\n', '')
+        diff = printable_diff.encode("utf-8") if print_diff else '<suppressed>'
+        diff_str = f'{bcolors.OKGREEN}Diff: {diff}{bcolors.ENDC}'
+
+    output = f'''
+    ~~~~~~~~~~~~~~~~~~~~~
+    {reason}
+    {date_str}
+    {hash_str}
+    {file_path}
+    {branch_str}
+    {lines_str}
+    {detail_str}
+    {commit_str[0:65]}
+    {diff_str}
+    ~~~~~~~~~~~~~~~~~~~~~
+    '''
+
+    print(output)
 
 
 def find_entropy(args, printable_diff, commit_time, branch_name, prev_commit, blob, print_diff):
     strings_found = []
+    line_numbers_found = []
     threshold = args.length_threshold
-    lines = printable_diff.split("\n")
 
-    for line in lines:
+    hunk_line_numbers = re.findall(r'@@ -(\d+),(\d+) \+(\d+),(\d+)', printable_diff)
```

Comment:
issue: I've seen cases where you don't have both sets of values. When a file is added or removed, for example. And this regex won't catch it.

Reply:
Fixed, I missed this. In future we should move the containing function out so that both the entropy and regex use the same function but call a different searcher. `search(f(x), args ....)` where f(x) is function calling entropy, regex, or some other different technique.